lib/adhearsion/events.rb

```diff
@@ -79,10 +78,24 @@ def refresh!
       end
 
       def clear
-        @queue = nil
+        kill!
+        @_executor = nil
         Handler.instance.clear_handlers
       end
+
+      def kill!
+        @_executor.kill if @_executor
+      end
+
+      def stop!
+        @_executor.shutdown if @_executor
+      end
```

Comment:
Looks like `stop!` is not currently used.  Though we can employ it in the future to smooth-out a graceful shutdown.

Reply:
exactly, this would be the preferred way to do a shutdown.
we can do eventually better to also wait for completion but if smt gets stuck that might be a dead-lock. that is the main reason I hooked up `kill!` which is more aggressive, not sure now if that is necessary.

here's from the native docs on `executor.shutdown` : 

> Initiates an orderly shutdown in which previously submitted tasks are executed, but no new tasks will be accepted. Invocation has no additional effect if already shut down.
>
> This method does not wait for previously submitted tasks to complete execution.